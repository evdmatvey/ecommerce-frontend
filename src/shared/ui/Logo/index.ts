export { default } from './ui/Logo';
