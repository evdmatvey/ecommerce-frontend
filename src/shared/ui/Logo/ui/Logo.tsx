import clsx from 'clsx';
import Image from 'next/image';
import { FC } from 'react';
import styles from './Logo.module.css';

interface LogoProps {
  withoutIcon?: boolean;
  size?: 'large' | 'standard';
}

const Logo: FC<LogoProps> = ({
  withoutIcon = false,
  size = 'standard',
}: LogoProps) => {
  const classes = clsx({
    [styles.withoutIcon]: withoutIcon,
    [styles.withIcon]: !withoutIcon,
    [styles.large]: size === 'large',
    [styles.standard]: size === 'standard',
  });
  return (
    <Image
      width={141}
      height={40}
      src={withoutIcon ? '/logo.svg' : '/icon-logo.svg'}
      className={classes}
      alt="cartzilla logo"
    />
  );
};

export default Logo;
