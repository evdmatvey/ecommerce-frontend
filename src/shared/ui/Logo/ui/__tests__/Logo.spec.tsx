import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Logo from '../Logo';

describe('Logo', () => {
  it('Standard with icon should render correctly', () => {
    const { asFragment } = render(<Logo size="standard" withoutIcon />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Standard without icon should render correctly', () => {
    const { asFragment } = render(<Logo size="standard" withoutIcon={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Large with icon should render correctly', () => {
    const { asFragment } = render(<Logo size="large" withoutIcon />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Large without icon should render correctly', () => {
    const { asFragment } = render(<Logo size="large" withoutIcon={false} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
