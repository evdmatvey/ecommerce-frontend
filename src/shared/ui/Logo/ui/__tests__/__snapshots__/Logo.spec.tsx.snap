// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > Large with icon should render correctly 1`] = `
<DocumentFragment>
  <img
    alt="cartzilla logo"
    class="_withoutIcon_4ce350 _large_4ce350"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/logo.svg"
    style="color: transparent;"
    width="141"
  />
</DocumentFragment>
`;

exports[`Logo > Large without icon should render correctly 1`] = `
<DocumentFragment>
  <img
    alt="cartzilla logo"
    class="_withIcon_4ce350 _large_4ce350"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/icon-logo.svg"
    style="color: transparent;"
    width="141"
  />
</DocumentFragment>
`;

exports[`Logo > Standard with icon should render correctly 1`] = `
<DocumentFragment>
  <img
    alt="cartzilla logo"
    class="_withoutIcon_4ce350 _standard_4ce350"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/logo.svg"
    style="color: transparent;"
    width="141"
  />
</DocumentFragment>
`;

exports[`Logo > Standard without icon should render correctly 1`] = `
<DocumentFragment>
  <img
    alt="cartzilla logo"
    class="_withIcon_4ce350 _standard_4ce350"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/icon-logo.svg"
    style="color: transparent;"
    width="141"
  />
</DocumentFragment>
`;
